src/vm/eventtrace.cpp

```diff
@@ -6332,13 +6337,63 @@ VOID ETW::MethodLog::SendMethodEvent(MethodDesc *pMethodDesc, DWORD dwEventOptio
     if(pMethodDesc->GetMethodTable_NoLogging())
         bIsGenericMethod = pMethodDesc->HasClassOrMethodInstantiation_NoLogging();
 
-    ulMethodFlags = ((ulMethodFlags |
+    int jitOptimizationTier = -1;
+    NativeCodeVersionId nativeCodeId = 0;
+    ulMethodFlags = ulMethodFlags |
         (bHasSharedGenericCode ? ETW::MethodLog::MethodStructs::SharedGenericCode : 0) |
         (bIsGenericMethod ? ETW::MethodLog::MethodStructs::GenericMethod : 0) |
         (bIsDynamicMethod ? ETW::MethodLog::MethodStructs::DynamicMethod : 0) |
-        (bIsJit ? ETW::MethodLog::MethodStructs::JittedMethod : 0) |
-        (bProfilerRejectedPrecompiledCode ? ETW::MethodLog::MethodStructs::ProfilerRejectedPrecompiledCode : 0) |
-        (bReadyToRunRejectedPrecompiledCode ? ETW::MethodLog::MethodStructs::ReadyToRunRejectedPrecompiledCode : 0)));
+        (bIsJit ? ETW::MethodLog::MethodStructs::JittedMethod : 0);
+    if (pConfig != nullptr)
+    {
+        if (pConfig->ProfilerRejectedPrecompiledCode())
+        {
+            ulMethodFlags |= ETW::MethodLog::MethodStructs::ProfilerRejectedPrecompiledCode;
+        }
+        if (pConfig->ReadyToRunRejectedPrecompiledCode())
+        {
+            ulMethodFlags |= ETW::MethodLog::MethodStructs::ReadyToRunRejectedPrecompiledCode;
+        }
+
+        _ASSERTE(((ulMethodFlags >> MethodFlagsJitOptimizationTierShift) & MethodFlagsJitOptimizationTierLowMask) == 0);
+#ifdef FEATURE_TIERED_COMPILATION
+        if (pMethodDesc->IsEligibleForTieredCompilation())
+        {
+            switch (pConfig->GetCodeVersion().GetOptimizationTier())
+            {
+                case NativeCodeVersion::OptimizationTier0:
+                    jitOptimizationTier = (int)JitOptimizationTier::QuickJitted;
+                    break;
+
+                case NativeCodeVersion::OptimizationTier1:
+                    jitOptimizationTier = (int)JitOptimizationTier::Optimized;
+                    break;
+
+                default:
+                    UNREACHABLE();
+            }
+        }
+#endif
+
+#ifdef FEATURE_CODE_VERSIONING
+        nativeCodeId = pConfig->GetCodeVersion().GetVersionId();
+#endif
+    }
+
+    if (jitOptimizationTier < 0)
+    {
+        if (pMethodDesc->IsJitOptimizationDisabled())
+        {
+            jitOptimizationTier = (int)JitOptimizationTier::MinOptJitted;
+        }
+        else
+        {
+            jitOptimizationTier = (int)JitOptimizationTier::Optimized;
+        }
+    }
+    static_assert_no_msg((unsigned int)JitOptimizationTier::Count - 1 <= MethodFlagsJitOptimizationTierLowMask);
+    _ASSERTE((unsigned int)jitOptimizationTier <= MethodFlagsJitOptimizationTierLowMask);
+    ulMethodFlags |= (unsigned int)jitOptimizationTier << MethodFlagsJitOptimizationTierShift;
```

Comment:
I thought I would keep it simple to start with :), the current info is based on `MethodDesc::IsJitOptimizationDisabled()`, which certainly doesn't cover all of the MinOpt cases. I'm looking into changing how it works, an issue is that we don't have a storage location for when the JIT decides to use `MinOpt`. That could potentially be solved by exposing `OptimizationTier` for methods that don't participate in tiering (with this change, `IsJitOptimizationDisabled()` returning true implies that the method is not tiered). I think I'll do something in-between for now, as there's not much time left. I'm thinking it would be fairly straightforward to reflect what the JIT decided to do RE `MinOpt`, but only when PerfView catches the runtime event - rundown events would not accurately reflect the MinOpt choice outisde of `IsJitOptimizationDisabled()`, so attaching to (actually detaching from) a running process would not get you accurate `MinOpt` info under the aforementioned condition for methods that were already jitted prior to attaching. I think that's good enough for now, definitely room for improvement here.